### PR TITLE
Feature/naviunni instructor prefs

### PR DESCRIPTION
**Pull request by Naviunni**
Modified the parsing and the view to be able to store instructor preferences
* Parsed the instructor csv to store preferences in a InstructorPreferences table
* Wrote some adhoc logic to populate name, email id etc if missing - to be removed later
* Modified the Instructor view to make it similar to the other views and added sorting
* Created a "View Preferences" button in the instructor view which displays preferences
* Improved the UI of the upload screen a bit
* Improved cucumber test coverage from 83% to 94% by adding features for the time slot view

Test coverage: 98.56%
Rubocop : 25 offenses

### CODE REVIEW

File: app/controllers/schedules_controller.rb [hunk 1]
```diff
@@ -12,7 +12,17 @@ def index
                                   "%#{params[:search_by_name]}%")
   end
 
-  def show; end
+  def show
+    Time.zone = 'America/Chicago'
```
Comment (wahibkapdi) [inline app/controllers/schedules_controller.rb:16]: Do we want to hard code the time zone here?

File: app/models/course.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # frozen_string_literal: true
 
+# Course model
 class Course < ApplicationRecord
   belongs_to :schedule
   has_many :sections
+  has_many :instructor_preferences
```
Comment (wahibkapdi) [inline app/models/course.rb]: Make this dependent: :destroy.

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -19,6 +19,8 @@
     <%= javascript_importmap_tags %>
     <%= javascript_include_tag 'room_bookings' %>
   </head>
+  <script src="https://cdn.jsdelivr.net/npm/@popperjs/core@2.11.6/dist/umd/popper.min.js"></script>
```
Comment (wahibkapdi) [inline app/views/layouts/application.html.erb:22]: What do we need the bootstrap js files for?

File: app/controllers/schedules_controller.rb (hunk 1 above)
Comment (Naviunni) [reply app/controllers/schedules_controller.rb:16]: I thought about it..but it's a very TAMU specific app, so why not

File: app/models/course.rb (hunk 2 above)
Comment (Naviunni) [reply app/models/course.rb]: ok

File: app/views/layouts/application.html.erb (hunk 3 above)
Comment (Naviunni) [reply app/views/layouts/application.html.erb:22]: The view preferences button

File: app/models/course.rb (hunk 2 above)
Comment (Naviunni) [reply app/models/course.rb]: On that note...shouldn't all schedule dependents be destroyed on schedule destroy as well?
class Schedule < ApplicationRecord
  has_many :rooms
  has_many :instructors
  has_many :courses
  validates :schedule_name, :semester_name, presence: true
end

These ^^^

File: app/controllers/schedules_controller.rb (hunk 1 above)
Comment (wahibkapdi) [reply app/controllers/schedules_controller.rb:16]: What if the client is on a vacation and wants to work from there?

File: app/models/course.rb (hunk 2 above)
Comment (wahibkapdi) [reply app/models/course.rb]: Yes

File: app/controllers/schedules_controller.rb (hunk 1 above)
Comment (Naviunni) [reply app/controllers/schedules_controller.rb:16]: :') I hope not
I will figure out how to use a dynamic time zone with my next feature

**Review by wahibkapdi — APPROVED** [review]

**Pull request merged**